twap/as/time.go

```diff
@@ -0,0 +1,38 @@
+package as
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// TimeNow fetches the current UTC time from an external API.
+// In the future this will be implemented as a host function.
+func TimeNow() (time.Time, error) {
+	req := HostHTTPRequestInput{
+		Method: "GET",
+		URL:    "https://timeapi.io/api/time/current/zone?timeZone=UTC",
+	}
+	resp, err := HostFuncHTTPRequest(req)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("making http request: %w", err)
+	}
+
+	timeData := struct {
+		DateTime string `json:"dateTime"`
+	}{}
+
+	err = json.Unmarshal(resp.Body, &timeData)
+	if err != nil {
+		return time.Time{}, fmt.Errorf(
+			"unmarshaling time data: %w...%s",
+			err,
+			resp.Body,
```

Comment:
heads up, this is a byte array but it is using a string in the format.  that is the same as casting the byte array as a string (as opposed to %v, which prints the byte array directly)  Not saying it is wrong or right, just making sure you are doing what you intend.

Reply:
thanks